app/src/main/java/com/example/mpweather/MainActivity.kt

```diff
@@ -13,108 +13,95 @@ import com.android.volley.toolbox.StringRequest
+            Log.d("response", "TestingInsideResponse")
+            // Log.d("response", response);
+            //declare variable to store string variable
+            var output = ""
+            //create a json object from the string response
+            try {
+                val jsonResponse = JSONObject(response)
+                //Get Json array call weather from Json response
+                val jsonArray = jsonResponse.getJSONArray("weather")
+                //Get Json object from Json array
+                val jsonObjectWeather = jsonArray.getJSONObject(0)
+                //Get the value of string description from json object weather
+                val description = jsonObjectWeather.getString("description")
+                //Get Json object from main from Json response
+                val jsonObjectMain = jsonResponse.getJSONObject("main")
+                //Get the values of temp, pressure, and humidity from Json object main
+                val temp = jsonObjectMain.getDouble("temp")
+                val feelsLike = jsonObjectMain.getDouble("feels_like")
+                val pressure = jsonObjectMain.getInt("pressure").toFloat()
+                val humidity = jsonObjectMain.getInt("humidity")
+                //Get Json object called wind
+                val jsonObjectWind = jsonResponse.getJSONObject("wind")
+                //Get the string speed meters per second
+                val wind = jsonObjectWind.getString("speed")
+                //Get the json called clouds
+                val jsonObjectClouds = jsonResponse.getJSONObject("clouds")
+                //Get the value of string all it represents cloudiness in percentage
+                val clouds = jsonObjectClouds.getString("all")
+                //Get the Json object called sys
+                val jsonObjectSys = jsonResponse.getJSONObject("sys")
+                //Get the value of the string country
+                val countryCode = jsonObjectSys.getString("country")
+                //Get name of city from Json response
+                val cityName = jsonResponse.getString("name")
+                //Change the color of text view
+                tvResult!!.setTextColor(Color.rgb(68, 134, 199))
+                // Create the output string by concatenating the necessary variables properly
+                output += """Current weather of $cityName ($countryCode)
+                        Temp: $temp  °C
+                        Feels Like: $feelsLike  °C
+                        Humidity: $humidity  %
+                        Description: $description
+                        Wind Speed: ${wind}m/s (meters per second)
+                        Cloudiness: $clouds%
+                        Pressure: $pressure  hPa"""
+                // Set text with output
+                tvResult!!.text = output
+            } catch (e: JSONException) {
+                e.printStackTrace()
```

Comment:
Use proper logging, and handle the exception properly.
As this exists, when something goes wrong:
1) The catch block will catch the exception.
2) It will print the stack trace, which no one will see.
3) Then, it will assume that the program executed correctly, and continue running to the next step.  Reason being, there's nothing stopping the program once the exception is caught.